core/templates/dev/head/exploration_editor/EditorServices.js

```diff
@@ -880,32 +990,23 @@ oppia.factory('explorationStatesService', [
 ]);
 
 oppia.factory('statePropertyService', [
-  '$log', 'changeListService', 'alertsService',
-  function($log, changeListService, alertsService) {
+  '$log', 'changeListService', 'alertsService', 'explorationStatesService',
+  function($log, changeListService, alertsService, explorationStatesService) {
     // Public base API for data services corresponding to state properties
-    // (interaction id, content, etc.) Note that this does not update
-    // explorationStatesService; it is maintained only locally.
+    // (interaction id, content, etc.)
     // WARNING: This should be initialized only in the context of the state
     // editor, and every time the state is loaded, so that proper behavior is
     // maintained if e.g. the state is renamed.
-    // TODO(sll): Make this update explorationStatesService.
+    // TODO(sll): Remove this service and its descendants, in favour of using
+    // explorationStatesService directly.
     return {
-      init: function(stateName, value, statesAccessorDict, statesAccessorKey) {
-        if (!statesAccessorDict || !statesAccessorKey) {
-          throw 'Not enough args passed into statePropertyService.init().';
+      init: function(stateName, value) {
+        if (this.setterMethodKey === null) {
+          throw 'State property setter method key cannot be null.';
         }
 
-        if (this.propertyName === null) {
-          throw 'State property name cannot be null.';
-        }
-
-        $log.info('Initializing state ' + this.propertyName + ':', value);
+        $log.info('Initializing state ' + this.setterMethodKey + ':', value);
```

Comment:
Do we still need this around? I thought we didn't want to log anything to the console.